app/src/main/java/com/github/houseorganizer/houseorganizer/SubTaskAdapter.java

```diff
@@ -30,18 +28,17 @@ public void onBindViewHolder(@NonNull BiViewHolder<Button, EditText> holder, int
         EditText titleEditor = holder.rightView;
 
         /* Setup for subtask title change */
-        TaskView.setUpSubTaskView(parentTask.getSubTaskAt(position), titleEditor);
+        TaskView.setUpSubTaskView(parentTask, position, titleEditor);
 
         /* Setup to mark subtask as done */
         doneButton.setOnClickListener(
                 v -> {
-                    parentTask.getSubTaskAt(position).markAsFinished();
+                    //parentTask.getSubTaskAt(position).markAsFinished();
+                    parentTask.removeSubTask(position);
                     new AlertDialog.Builder(v.getContext())
```

Comment:
I don't know if there is anything else but AlertDialog sounds like something for exception no ? Again I may be saying nonsense

Reply:
Yeah, right now I'm using it as a quick way to make things pop up, but I can change this to another View when refining the UI.

Reply:
I think AlertDialog means popup dialog from what i understand, its been used for adding and editing events among other things.